### DISCUSSION TITLE
Lost config file (how to regenerate it)?

### DISCUSSION THREAD

**Discussion by michealp-coder** [tailwindlabs/tailwindcss#5400] [Help]
**Summary**: I took over managing a TailwindCSS website and don't have the `config.js` file.

**Problem/Question**: is there a way to reverse generate the`config.js` file itself that was used to bundle the TailwindCSS on my website?

**Detail**: I joined a company where the person I replaced (who left the company), built our site using TailwindCSS. The problem is, I can't find the tailwind.config file used to build the Tailwind CSS bundle.

**What have I tried**: I have tried loading up a single page of our corporate site at https://play.tailwindcss.com/ and iterating (manually) on what the `config.js` file might have included that generated our CSS; however, the output is radically off.

_As such, is there a technical way to "reverse generate" what the `tailwind.config.js` was that resulted in the CSS that generated my site?_

Thanks so much in advance (truly).

**Comment by clintongreen** [comment]
This sounds really tough mate, really interested to see if you can reverse generate it.
One idea though, possibly it wasnt customised at all and thats why there isnt a config file.
Another one is to look at the tailwind folder in node_modules possibly you'll find something in the dist folder.

**Reply by michealp-coder** [reply]
Unfortunately, it’s definitely customized quite a bit. 

There’s nothing left of the original build process :(

Any other ideas?

**Reply by michealp-coder** [reply]
Is there a way to:

a. Download the non-purged default- tailwind.css file

b. Diff (at the class level not file level) my-tailwind.css with default-tailwind.css

c. Remove from the diff any css classes found in default-tailwind that aren’t in my-tailwind

d. Remove any class that are the same between both files

e. (Which should leave) only classes that are new or altered from the default tailwind file

Since this would be a diff at the class level (not file level) I’m not sure how to do that. 

Thoughts.

**Reply by michealp-coder** [reply]
I'm going to attempt to do the following:

a. use the full tailwind.css (non-purged) file found at https://tailwindcss.com/docs/installation#using-tailwind-via-cdn

b. sort the classes alphabetically be name

c. sort the attributes alphabeticvally by name

d. do (b) and (c) on my-tailwind.css

e. then diff the two

This should show all differences between the full-blown tailwind.css and my-purged-and-config version.

It will be a lot of manual effort since I need to manually exclude all of the excess classes that tailwind.css has that my-tailwind.css doesn't have but hopefully I'll have a diff'ed version to compare that will allow me to see what's different (which should give insights into what was included in the config file).

**Reply by michealp-coder** [reply]
I'll report back my progress.

**Reply by mgav** [reply]
I'm 99.9% sure you've already done this, but consider looking through any regular backups not related to this code, such as Time Machine for Mac. 

Also, is there any way to search outgoing email attachments from the person you replaced, just in case? 

Good luck!

**Reply by simonswiss** [reply]
Ah man what a rough situation! How large is the project? Maybe another approach (less methodical than what you propose) would be to do a project-wide search for a common class pattern (like `px-` or `font-`), and see in the results if you find anything that seems out of the ordinary..

I can't really think of any way to "detect" whatever class is out of the default generated CSS by Tailwind.

You have absolutely zero ways to contact that person that you've replaced? The company may even put a bounty on him providing that file, considering how many hours you may spend on company time trying to reverse engineer this config..

**Comment by michealp-coder** [comment]
Really appreciate your suggestion and help. I'll give it a try.

**Comment by michealp-coder** [comment]
I really appreciate your help and suggestion. I'll give it a try. Thanks so
much.
